js/src/forum/states/FlagListState.js

```diff
@@ -0,0 +1,38 @@
+export default class FlagListState {
+  constructor(app) {
+    this.app = app;
+
+    /**
+     * Whether or not the flags are loading.
+     *
+     * @type {Boolean}
+     */
+    this.loading = false;
+
+    this.cache = [];
+  }
+
+  /**
+   * Load flags into the application's cache if they haven't already
+   * been loaded.
+   */
+  load() {
+    if (this.cache.length && !app.session.user.attribute('newFlagCount')) {
```

Comment:
`app` should be `this.app`, same a few lines below.

Personally I don't see much interest actually saving `app` as `this.app` though. There won't be different app objects, we could just access it via the global scope instead of having it as a `FlagListState` constructor parameter.

Reply:
That's a good catch, but shouldn't we be consistent with the approach taken in core re: globals?

Reply:
I feel like the global stuff is going a bit too far. while there were some reasons to have for example multiple discussion list states exist in the app at one point, I see no reason to have more than one `app` or `flags` object in the app at any point, so I'd skip passing them as prop completely and always access them from global.

It would only make sense to pass everything as props in the context of unit testing, but we're not unit testing Mithril components so :man_shrugging:

Reply:
I'll defer to @franzliedke on this one as he's our main frontend DI champion